src/elem.rs

```diff
@@ -73,6 +73,24 @@ where
     }
 }
 
+pub trait Abs {
+    type Output;
+
+    #[must_use]
+    fn abs(self) -> Self::Output;
+}
+
+impl<T: Float> Abs for Elem<T> {
+    type Output = Elem<T>;
+
+    fn abs(self) -> Elem<T> {
+        match self {
+            Elem::Image(image) => Elem::Image(image.mapv(|e| e.abs())),
+            Elem::Value(value) => Elem::Value(value.abs()),
```

Comment:
You can't really do that for this operator, but for the other 4 operators, you should probably write your own macro. Something like:

    bin_op!(Mul, mul, *)
    bin_op!(Div, div, /)

I'm not entirelly sure that it's possible to write a single operator like `/` though :) Maybe the macro system accepts it.

Reply:
This works really well.

Reply:
Also I could make macro for all unary operation provided in https://doc.rust-lang.org/std/primitive.f64.html.